lib/src/revset.rs

```diff
@@ -1609,11 +1626,17 @@ fn resolve_commit_ref(
         RevsetCommitRef::RemoteBranches {
             branch_pattern,
             remote_pattern,
+            remote_ref_state,
         } => {
             // TODO: should we allow to select @git branches explicitly?
             let commit_ids = repo
                 .view()
                 .remote_branches_matching(branch_pattern, remote_pattern)
+                .filter(|(_, remote_ref)| {
+                    remote_ref_state
+                        .map(|state| remote_ref.state == state)
+                        .unwrap_or(true)
```

Comment:
nit: `.map_or(true,` also works (if you like)

Reply:
Oh I didn't know about that function, that's nice!